### PR TITLE
docs(zone.js): update docs to enable beforeunload

### LINKED ISSUE #52256: Angular 16 @HostListener('window:beforeunload', ['$event']) is not working
### Which @angular/* package(s) are the source of the bug?

Zone.js

### Is this a regression?

No

### Description

After migration from 15.2.10 to 16.2.* (tried 16.2.3 and 16.2.9) the working approach with showing the default browser preventing reload popup is not working any more.
image

Working minimalistic solution (Angular 15) https://stackblitz.com/edit/angular-15-starter-pack-iuvmnk
The same solution with Angular 16 (not working) https://stackblitz.com/edit/stackblitz-starters-qhmmsm

### Step to reproduce
-  open https://stackblitz.com/edit/stackblitz-starters-qhmmsm with Chromium based browser (Chrome/Edge)
- wait until the app is built
- open console
- click between the header and the link (you will see 'Changed!' output in the console)
- try to reload the page (either the embedded in stackblitz or the browser page - it doesn't matter, should work the same in both cases)

ER: the default browser preventing reload popup should appear
AR: the default browser preventing reload popup doesn't appear

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/stackblitz-starters-qhmmsm

### Please provide the exception or error you saw

```true
The browser doesn't show the default preventing reload popup.
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
@angular/core@16.2.9
@angular/common@16.2.9
@angular/animations@16.2.9
@angular/compiler@16.2.9
@angular/forms@16.2.9
@angular/platform-browser@16.2.9
@angular/platform-browser-dynamic@16.2.9

Chrome 118.0.5993.89 64-bit / Chrome 118.0.5993.71 64-bit / Chrome 117.0.5938.132 64-bit / Edge 117.0.2045.60 64-bit
```

### Anything else?

As you can see from the component code if I click on the layout and change it to the changed state I shoud see the default browser popup asking "Reload the site?", but in Chrome 118.0.5993.89 64-bit / Chrome 118.0.5993.71 64-bit / Chrome 117.0.5938.132 64-bit / Edge 117.0.2045.60 64-bit it doesn't appear (meanwhile it works in Firefox 118.0.2 64-bit or 15.2.10 version), Win 10.
One more mention about the same problem here: https://github.com/angular/angular/issues/50935

### PR DESCRIPTION
**Pull request by arturovt**
In this commit, we update the documentation to reflect the property that allows enabling the default browser `beforeunload` handling, which was added in a previous commit.

Additionally, some cosmetic grammar changes have been made in this documentation, as the previous text had some issues.

Closes #52256

### CODE REVIEW

**Comment by JeanMeche** [comment]
It looks like the rendering of the markdown is broken because of indentation, can you take a look at it ? Thank you. 

Screenshot 2024-09-19 at 19 14 16

**Comment by arturovt** [comment]
@JeanMeche updated.

File: packages/zone.js/MODULE.md [hunk 1]
```diff
@@ -60,50 +61,62 @@ Below is the full list of currently supported modules.
...
+    },
+  ];
+</script>
+<script src="../bundles/zone.umd.js"></script>
+```
 
-You can also disable specific on_properties by setting `__Zone_ignore_on_properties` as follows: for example,
-if you want to disable `window.onmessage` and `HTMLElement.prototype.onclick` from zone.js patching,
-you can do like this.
+Ignoring `on` properties from being patched means that callbacks will always be invoked within the root context, regardless of where the `on` callback has been set. Even if `onclick` is set within a child zone, the callback will be called inside the root zone:
```
Comment (AndrewKushnir) [inline packages/zone.js/MODULE.md:91]:
```suggestion
Excluding `on` properties from being patched means that callbacks will always be invoked within the root context, regardless of where the `on` callback has been set. Even if `onclick` is set within a child zone, the callback will be called inside the root zone:
```

File: packages/zone.js/MODULE.md [hunk 2]
```diff
@@ -60,50 +61,62 @@ Below is the full list of currently supported modules.
...
+```ts
+Zone.current.fork({ name: 'child' }).run(() => {
+  document.body.onclick = () => {
+    console.log(Zone.current); // <root>
+  };
+});
 ```
 
-- Error
+For unpatching events being added through `addEventListener`, please refer to [UnpatchedEvents](./STANDARD-APIS.md#unpatchedevents).
```
Comment (AndrewKushnir) [inline packages/zone.js/MODULE.md:101]:
```suggestion
You can find more information on adding unpatched events via `addEventListener`, please refer to [UnpatchedEvents](./STANDARD-APIS.md#unpatchedevents).
```

File: packages/zone.js/MODULE.md [hunk 3]
```diff
@@ -60,50 +61,62 @@ Below is the full list of currently supported modules.
...
-This package will provide following functionality.
+### Error
 
-  1. Error inherit: handle `extend Error` issue.
-     ```
-       class MyError extends Error {}
-       const myError = new MyError();
-       console.log('is MyError instanceof Error', (myError instanceof Error));
-     ```
+By default, `zone.js/plugins/zone-error` will not be loaded due to performance concerns.
```
Comment (AndrewKushnir) [inline packages/zone.js/MODULE.md:105]:
```suggestion
By default, `zone.js/plugins/zone-error` will not be loaded for performance reasons.
```

File: packages/zone.js/MODULE.md [hunk 4]
```diff
@@ -60,50 +61,62 @@ Below is the full list of currently supported modules.
...
-     without `zone-error` patch, the example above will output `false`, with the patch, the result will be `true`.
+1. **Error Inheritance:** Handle the `extend Error` issue:
+
+   ```ts
+     class MyError extends Error {}
+     const myError = new MyError();
+     console.log('is MyError instanceof Error', (myError instanceof Error));
+   ```
+
+   Without the `zone-error` patch, the example above will output `false`; with the patch, the result will be `true`.
```
Comment (AndrewKushnir) [inline packages/zone.js/MODULE.md:116]:
```suggestion
   Without the `zone-error` patch, the example above will output `false`. With the patch, the result will be `true`.
```

**Review by AndrewKushnir — COMMENTED** [review]
@arturovt thanks for documentation updates! Just left a few suggestions.

**Review by JiaLiPassion — APPROVED** [review]
LGTM!

**Comment by AndrewKushnir** [comment]
This PR was merged into the repository by commit 8ce190b1b7a12c024b4c333c11b2fb119f93c012.

The changes were merged into the following branches: main, 18.2.x

**Pull request closed without merging**